files/class-api-client.php

```diff
@@ -145,8 +145,9 @@ public function upload_file( $local_path, $upload_path ) {
 		// save to cache
 		$this->cache->copy_to_cache( $response_data->filename, $local_path );
 
-		// reset file stats cache if any
-		$this->cache->remove_stats( $response_data->filename );
+		// Reset file stats cache, if any.
+		// Note: the ltrim is because we store the path without the leading slash but the API returns the path with it.
+		$this->cache->remove_stats( ltrim( $response_data->filename, '/' ) );
```

Comment:
Maybe it could have been easier to just use the `$upload_path`. In the end, that's what we generate internally in the wrapper for everything, so it's better that the cache matches it exactly.

Reply:
Oh, yes, that would definitely be much cleaner :)